tests/unit/websocket_test.cc

```diff
@@ -62,5 +63,22 @@ SEASTAR_TEST_CASE(test_websocket_handshake) {
         for (auto& header : resp->_headers) {
             std::cout << header.first << ':' << header.second << std::endl;
         }
+
+        dummy.register_handler("echo", [] (input_stream<char>& in,
+                output_stream<char>& out) {
+            return make_ready_future<>();
+        });
+
+        const std::string frame =
+            "\x01\x8C\x00\x00"
+            "\x00\x00\x48\x65"
+            "\x6c\x6c\x6f\x20"
+            "\x77\x6f\x72\x6c"
+            "\x64\x21";
+        const std::string response = frame;
+        
+        // TODO: actually write the test.
```

Comment:
Looks like you're almost there - just register the same handler as in the demo program and verify that you can read exactly what you wrote

Reply:
I am not sure why I cannot add you as a reviewer, but some further modifications of websocket_test are in this pull request:
https://github.com/psarna/seastar/pull/110